test/cluster/config_test.go

```diff
@@ -0,0 +1,419 @@
+// - add Root CA & Intermediate CA to CertAuthConfig
+func TestUpdateCA(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+	leaderServer := c.Servers[leaderIndex]
+
+	//update to empty CA returns error
+	configEnv, err := leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig := configEnv.GetResponse().GetConfig()
+	newConfig.CertAuthConfig = nil
+	version := configEnv.GetResponse().GetMetadata().GetVersion()
+
+	_, _, err = leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.EqualError(t, err, "failed to submit transaction, server returned: status: 400 Bad Request, message: Invalid config tx, reason: CA config is empty. At least one root CA is required")
+
+	// add a Root CA & Intermediate CA
+	certRootCA, caPrivKey, err := testutils.GenerateRootCA("Orion RootCA", "127.0.0.1")
+	require.NoError(t, err)
+	require.NotNil(t, certRootCA)
+	require.NotNil(t, caPrivKey)
+	rootCAkeyPair, err := tls.X509KeyPair(certRootCA, caPrivKey)
+	require.NoError(t, err)
+	require.NotNil(t, rootCAkeyPair)
+	certIntCA, _, err := testutils.GenerateIntermediateCA("Orion IntermediateCA", "127.0.0.1", rootCAkeyPair)
+	require.NoError(t, err)
+	require.NotNil(t, certIntCA)
+
+	configEnv, err = leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig = configEnv.GetResponse().GetConfig()
+	caConf := newConfig.CertAuthConfig
+	decodedCertRootCA, _ := pem.Decode(certRootCA)
+	caConf.Roots = append(caConf.Roots, decodedCertRootCA.Bytes)
+	decodedCertIntCA, _ := pem.Decode(certIntCA)
+	caConf.Intermediates = append(caConf.Intermediates, decodedCertIntCA.Bytes)
+
+	version = configEnv.GetResponse().GetMetadata().GetVersion()
+	txID, rcpt, err := leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.NoError(t, err)
+	require.NotNil(t, rcpt)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt)
```

Comment:
Now on-board a user whos certificate was generated by the intemediate CA, and check that the server accepts her.